### PR TITLE
Add string function split_part

### PR DESCRIPTION
**Pull request by Xuxue1**
Add string function split_part. Just like postgresql

### CODE REVIEW

File: be/src/exprs/string_functions.cpp [hunk 1]
```diff
@@ -761,4 +761,61 @@ StringVal StringFunctions::money_format(FunctionContext *context, const LargeInt
...
+            }
+        }
+    }
+    return -1;
+}
+
+
+StringVal StringFunctions::split_part(FunctionContext* context,const StringVal& content,
+                                      const StringVal& delimiter,const IntVal& field) {
+    if (field.val <= 0) return StringVal::null();
```
Comment (chenhao7253886) [inline be/src/exprs/string_functions.cpp]: What if content、delimiter or field is null？

File: be/src/exprs/string_functions.cpp [hunk 2]
```diff
@@ -761,4 +761,61 @@ StringVal StringFunctions::money_format(FunctionContext *context, const LargeInt
...
+    int find[field.val]; //store substring position
+    for(int i=0;i<=field.val;i++) find[i] = -1; // init
+    int from = 0;
+    for(int i=1;i<=field.val;i++){ // find
+        find[i-1] = indexOf(content.ptr,0,content.len, delimiter.ptr,0,delimiter.len,from);
```
Comment (imay) [inline be/src/exprs/string_functions.cpp]:
```suggestion
        find[i-1] = indexOf(content.ptr,0, content.len, delimiter.ptr, 0, delimiter.len, from);
```
You should add one space after punctuation

File: be/src/exprs/string_functions.cpp [hunk 3]
```diff
@@ -761,4 +761,61 @@ StringVal StringFunctions::money_format(FunctionContext *context, const LargeInt
...
+        from = find[i-1] + 1;
+        if (find[i-1] == -1) {
+            break;
+        }
+    }
+    if ((field.val>1 && find[field.val-2] == -1) || (field.val==1 && find[field.val-1] == -1)){ // not find
```
Comment (imay) [inline be/src/exprs/string_functions.cpp]:
```suggestion
    if ((field.val > 1 && find[field.val-2] == -1) || (field.val == 1 && find[field.val-1] == -1)){ // not find
```

File: be/src/exprs/string_functions.cpp [hunk 4]
```diff
@@ -761,4 +761,61 @@ StringVal StringFunctions::money_format(FunctionContext *context, const LargeInt
...
+        return StringVal::null();
+    }
+    int start_pos,len;
```
Comment (imay) [inline be/src/exprs/string_functions.cpp]: define `len` when it is needed

File: be/src/exprs/string_functions.cpp [hunk 5]
```diff
@@ -761,4 +761,61 @@ StringVal StringFunctions::money_format(FunctionContext *context, const LargeInt
     return do_money_format(context, ss.str());
 }
 
+static int indexOf(const uint8_t* source, int sourceOffset, int sourceCount,
```
Comment (imay) [inline be/src/exprs/string_functions.cpp]:
```suggestion
static int index_of(const uint8_t* source, int sourceOffset, int sourceCount,
```

**Review by imay — CHANGES_REQUESTED** [review]

**Comment by Xuxue1** [comment]
感谢指出问题， 我仔细修改下再提交。

**Pull request closed without merging**